### ISSUE TITLE
[BUG][SDW]Aplay/ arecord isn't continued after switch to background then switch back foreground.

### ISSUE THREAD

**Issue by ClarexZhou** [thesofproject/linux#1220]
**Describe the bug**
Aplay/ arecord isn't continued after switch to background then switch back foreground.

**To Reproduce**
1. aplay -Dhw:0,0 -r48000 -c2 -fs24_le wav/long.wav -vv -i 
2. press ctrl-z to put aplay background
3. enter 'fg' to switch aplay foreground.

**frequency**
100%

**Actual result**
Aplay isn't continued.

**Expected result**
Aplay should be continued normally.

**Environment**
linux: https://github.com/thesofproject/linux/commits/integration/soundwire-alpha commit: a8f3408
sof: https://github.com/thesofproject/sof/commits/sdw-001-drop-stable commit: 0223d94
Topology: file: sof-cml-rt700.m4 same path as firmware
platform: CML SDW RVP

No error messages.

[bg_fg_switch_fail_logger.txt](https://github.com/thesofproject/sof/files/3557994/bg_fg_switch_fail_logger.txt)
[bg_fg_switch_fail_dmesg.txt](https://github.com/thesofproject/sof/files/3557995/bg_fg_switch_fail_dmesg.txt)

**Comment by plbossart** [comment]
please retest with firmware master and kernel integration/soundwire-latest branch

**Comment by ClarexZhou** [comment]
Error seen with firmware master and kernel integration/soundwire-latest branch
**Environment**
Firmware: 460c41a  Master
Kernel: 4ac5440 on integration/soundwire-latest
Topology: file: same with firmware branch, sof-cml-rt700.m4
platform: CML rt700 SDW RVP

**Comment by plbossart** [comment]
ACK, behavior reproduced.

@bardliao is this something that can be solved with your PR https://github.com/thesofproject/linux/pull/1166

**Comment by bardliao** [comment]
> @bardliao is this something that can be solved with your PR #1166

Yes, the issue is what #1166 try to solve.

**Comment by YvonneYang2** [comment]
Issue reproduced when run ./sof-test-xrun.sh stress script with test recipe as below.

Firmware: 460c41aa https://github.com/thesofproject/sof/commits/master
Kernel: 4ac5440 https://github.com/thesofproject/linux/tree/integration/soundwire-latest
tplg: sof-icl-rt711-rt1308-rt715-hdmi.tplg rename sof-icl-rt711-rt1308-rt715.tplg https://github.com/bardliao/sof/blob/fix/soundwire-tplg/tools/topology/
platform: ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode

**dmesg**
```
[  120.036103] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[  120.036290] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60050000: GLB_STREAM_MSG: TRIG_STOP
[  120.036291] sof-audio-pci 0000:00:1f.3: ipc tx: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[  120.036456] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x60030000: GLB_STREAM_MSG: PCM_FREE
[  120.037848] rt711 sdw:0:25d:711:0:1: rt711_clock_config complete, clk_freq=4800000
[  120.038453] intel-sdw sdw-master-0: Compute bus params failed: -22
[  120.038455] intel-sdw sdw-master-0: Compute params failed: -22
[  120.038456] Prepare for stream:SDW0 Pin2-Playback failed: -22
[  120.038458] intel-sdw sdw-master-0: ASoC: cpu DAI prepare error: -22
[  120.038459]  SDW0-Playback: ASoC: backend prepare failed -22
[  120.038474] sof-audio-pci 0000:00:1f.3: pcm: free stream 0 dir 0
[  120.038619] sof-audio-pci 0000:00:1f.3: ipc tx: 0x80010000: GLB_DAI_MSG: CONFIG
[  120.038816] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x80010000: GLB_DAI_MSG: CONFIG
[  120.038862] sof-audio-pci 0000:00:1f.3: pcm: close stream 0 dir 0
```
[dmesg.log](https://github.com/thesofproject/linux/files/3633922/dmesg.log)
[logger.log](https://github.com/thesofproject/linux/files/3633923/logger.log)
[sof-test-xrun.txt](https://github.com/thesofproject/linux/files/3633934/sof-test-xrun.txt)

**Comment by plbossart** [comment]
please retest now that https://github.com/thesofproject/linux/pull/1239 was merged

**Comment by YvonneYang2** [comment]
Issue cannot reproduce on ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode. 

Test recipe:
Firmware: 0b47129 https://github.com/thesofproject/sof/commits/master + PR1851 
Kernel: e24f321 https://github.com/thesofproject/linux/tree/integration/soundwire-latest
tplg:  sof-icl-rt711-rt1308-rt715-hdmi.tplg rename sof-icl-rt711-rt1308-rt715.tplg  https://github.com/bardliao/sof/tree/fix/soundwire-topology/tools/topology
platform: ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode

**Comment by bardliao** [comment]
> Issue cannot reproduce on ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode.
> 
> Test recipe:
> Firmware: 0b47129 https://github.com/thesofproject/sof/commits/master + PR1851
> Kernel: [e24f321](https://github.com/thesofproject/linux/commit/e24f321b25c2d7ed4ee65bc9eebac2b62093cee8) https://github.com/thesofproject/linux/tree/integration/soundwire-latest
> tplg: sof-icl-rt711-rt1308-rt715-hdmi.tplg rename sof-icl-rt711-rt1308-rt715.tplg https://github.com/bardliao/sof/tree/fix/soundwire-topology/tools/topology
> platform: ICL RVP, with codec ALC711 -RT1308, RT715 in SDW mode

@YvonneYang2 So, can we close it?

**Comment by ClarexZhou** [comment]
On CML rtc700 still not work,aplay/ arecord isn't continued . But no any ipc message in dmesg once switch to foreground. As below:
![IMG_20190923_135231](https://user-images.githubusercontent.com/49641224/65407018-c4434a80-de13-11e9-9775-df622b3e61a4.jpg)

**Environment**
Firmware: 0b47129  Master
Kernel: e24f321b on integration/soundwire-latest
Topology: file: same with firmware branch, sof-cml-rt700.m4
platform: CML RVP with onboard ALC700 in SDW mode

**Comment by YvonneYang2** [comment]
Issue cannot reproduce on ICL RVP with onboard ALC700 in SDW mode.

Environment
Firmware: 0b47129 Master
Kernel: e24f321 on integration/soundwire-latest
Topology: file: same with firmware branch, sof-icl-rt700.m4
platform: ICL RVP with onboard ALC700 in SDW mode

**Comment by bardliao** [comment]
@ClarexZhou @YvonneYang2 The issue occurs in all sound devices on the CML RVP, including HDMI and DMIC. So it looks like not be a soundwire issue. Besides, there is no any dmesg output even I added trace in snd_pcm_common_ioctl().

**Comment by ClarexZhou** [comment]
That issue on CML rtc700 is SSD issue.

Close for fixed by #1239
